pkg/operator2/oauth.go

```diff
@@ -136,3 +148,173 @@ func (c *osinOperator) handleOAuthConfig(configOverrides []byte) (*corev1.Config
 		},
 	}, nil
 }
+
+func convertProviderConfigToOsinBytes(providerConfig *configv1.IdentityProviderConfig) []byte {
+	var providerConfigBytes bytes.Buffer
+
+	bytesWriter := bufio.NewWriter(&providerConfigBytes)
+	serializer := kubejson.NewYAMLSerializer(kubejson.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
+
+	switch providerConfig.Type {
```

Comment:
`case`s need to be in alphabetical order for my sanity.

Reply:
Sorted. I originally chose the same order as in the openshift/api version of `types_oauth.go`